src/hardware/mixer.cpp

```diff
@@ -1324,7 +1326,40 @@ std::unique_ptr<Program> MIXER_ProgramCreate() {
 	return ProgramCreate<MIXER>();
 }
 
-void MIXER_Init(Section* sec) {
+enum class MixerState : uint8_t {
+	Off,
+	On,
+	Unset,
+};
+
+static void SetMixerState(const MixerState requested_state)
+{
+	assert(requested_state != MixerState::Unset);
+
+	static auto current_state = MixerState::Unset;
+
+	if (current_state == requested_state)
+		return;
+
+	if (mixer.nosound && requested_state == MixerState::On) {
+		LOG_ERR("MIXER: Cannot enable the mixer while nosound is enabled");
+		return;
+	} else if (requested_state == MixerState::On) {
+		assert(!mixer.nosound);
+		TIMER_DelTickHandler(MIXER_Mix_NoSound);
+		TIMER_AddTickHandler(MIXER_Mix);
+		SDL_PauseAudioDevice(mixer.sdldevice, 0);
+	} else {
+		assert(requested_state == MixerState::Off);
+		TIMER_DelTickHandler(MIXER_Mix);
+		TIMER_AddTickHandler(MIXER_Mix_NoSound);
+		SDL_PauseAudioDevice(mixer.sdldevice, 1);
+	}
+	current_state = requested_state;
+}
```

Comment:
Wouldn't this achieve the same thing in a simpler way? I'm also unsure why we need the `Unset` state, so maybe we can just use a boolean for this?

This is just a static function used only in this file, and the state validation is already done at the call site. Or maybe you can move all the state validation and update logic from `ToggleMute` into here and that method would simply just call this.

```cpp
enum class MixerState { Off, On };

static void SetMixerState(const MixerState requested_state)
{
	switch (requested_state) {
	case MixerState::On:
		TIMER_DelTickHandler(MIXER_Mix_NoSound);
		TIMER_AddTickHandler(MIXER_Mix);
		SDL_PauseAudioDevice(mixer.sdldevice, 0);
		break;

	case MixerState::Off:
		TIMER_DelTickHandler(MIXER_Mix);
		TIMER_AddTickHandler(MIXER_Mix_NoSound);
		SDL_PauseAudioDevice(mixer.sdldevice, 1);
		break;
	}
}
```